File: mercury_v2_backend/app/schemas/gallery.py
from typing import List, Optional
from pydantic import BaseModel
from .design_brief import DesignBriefBase

class GalleryItem(BaseModel):
    id: str
    image_url: str
    title: str
    brief_data: DesignBriefBase

class GalleryResponse(BaseModel):
    items: List[GalleryItem]
    total: int

File: mercury_v2_frontend/src/types/api/gallery.ts
import type { DesignBrief } from './designBrief';

export interface ReferenceItem {
  id: string;
  image_url: string;
  title: string;
  brief_data: Partial<DesignBrief>;
}

export interface ReferenceGalleryResponse {
  items: ReferenceItem[];
  total: number;
}
